src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1960,9 +1988,20 @@ public StringBuilder Replace(string oldValue, string? newValue, int startIndex,
             {
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_IndexMustBeLessOrEqual);
             }
-            ArgumentException.ThrowIfNullOrEmpty(oldValue);
 
-            newValue ??= string.Empty;
+            if (oldValue == null)
+            {
+                throw new ArgumentNullException(nameof(oldValue));
+            }
```

Comment:
These null checks need to be in the string-specific overloads. Spans won't be null.

Reply:
Re-added the `ArgumentException.ThrowIfNullOrEmpty(oldValue);` call in the string-specific overload, set new value to `ReadOnlySpan<char>.Empty` if a `null` value is provided as `newValue` in the string-specific overload and removed the null checks in the `ReadOnlySpan<char>` implementation (and the corresponding case in the unittest).